pkg/zfs/volume.go

```diff
@@ -417,9 +417,9 @@ func GetUserFinalizers(finalizers []string) []string {
 
 // IsVolumeReady returns true if volume is Ready
 func IsVolumeReady(vol *apis.ZFSVolume) bool {
-
-	if vol.Status.State == ZFSStatusReady {
-		return true
+	if vol.Status.State != "" {
```

Comment:
Would be good to add comment here explaining why this is needed

Reply:
I have added a comment explaining why this is necessary. Please let me know if any further details are needed.

Reply:
Thanks, maybe clarify newer to mention the version from which the status was added

Reply:
Thank you for the suggestion. I have clarified the version from which the status was added to provide better context.